cmake/developer_package/compile_flags/os_flags.cmake

```diff
@@ -492,16 +492,3 @@ function(link_system_libraries TARGET_NAME)
         endif()
     endforeach()
 endfunction()
-
-#
-# ov_try_use_gold_linker()
-#
-# Tries to use gold linker in current scope (directory, function)
-#
-function(ov_try_use_gold_linker)
-    # gold linker on ubuntu20.04 may fail to link binaries build with sanitizer
-    if(CMAKE_COMPILER_IS_GNUCXX AND NOT ENABLE_SANITIZER AND NOT CMAKE_CROSSCOMPILING)
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fuse-ld=gold" PARENT_SCOPE)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fuse-ld=gold" PARENT_SCOPE)
-    endif()
```

Comment:
maybe we just need to adjust condition when gold linker is enabled?
If your custom linker does not support this option, we can check it here and skip for your specific use-case

Reply:
I think that the proper approach would be to have a `CMakePresets.json` file for OpenVINO project. It is a common way of storing configuration parameters for CMake projects. The occasional linker change in some parts of the projects seems like a band-aid solution rather than a proper solution. What do you think?

Reply:
`CMakePresets.json` serves to store global per-project configurations and it will not help in configuring of particular targets.

Reply:
BTW, what kind of benefits your custom linker provides?

Reply:
@ilya-lavrenov basically the speed. you can take a look at the official github page of the mold linker https://github.com/rui314/mold

Reply:
@ilya-lavrenov I've added the exception for the mold linker, please take a look